src/search/algorithms/int_packer.cc

```diff
@@ -83,6 +84,8 @@ void IntPacker::set(Bin *buffer, int var, int value) const {
 
 void IntPacker::pack_bins(const vector<int> &ranges) {
     assert(var_infos.empty());
+    assert(all_of(ranges.begin(), ranges.end(),
+                  [](int range) {return range > 1;}));
```

Comment:
Is a range of 1 really a problem here? I can imagine this but I didn't see it, at least. It should take 1 bit to represent a variable with a single value. Variables like this are not useful but this should not be the concern of this algorithmic class.

Is the issue that `get_bit_size_for_range(1)` returns 0? In that case, I think that function should be fixed instead.

Reply:
Good catch, done.

Reply:
I don't agree. Wasting memory on a variable that can only take on a single value is a usage error for this class that it would be good to point out to the user. The point of these classes is to represent stuff space-efficiently. Wasting a bit on something that can only ever have the same value runs against this.

Reply:
Which solution would you suggest instead?

Reply:
I have two possible suggestions, but I think it might be more efficient to quickly discuss this over Zoom if that works for you. I have a question or two about the use case that triggered the change. Are you available right now? If yes, let's arrange a Zoom call on Discord.

Reply:
Unfortunately, I should get back to finishing ICAPS workshop papers now :-) Maybe we can chat briefly about this after one of the next reading groups or Fast Downward meetings.

Reply:
No worries. The short version not knowing the background for the change is: the old code was intended for the assumption that domain sizes are at least 2. If the assumption is still valid, we could add an assertion about this to make this point more explicit (assuming there wasn't already such an assertion). If we do need support for variables with domain {0} because it's not OK to require the user of the code to filter out such variables, we should store them in 0 bits, not 1. I don't think this would be difficult to do.

Can we arrange a concrete time to discuss this? I hope it will be only 5 minutes. I don't know when the workshop deadline is, but for example tomorrow I have some gaps in the schedule.